apis/python/src/tiledbsoma/sparse_nd_array.py

```diff
@@ -177,32 +174,6 @@ def __init__(self, sr: clib.SOMAReader, shape: NTuple):
         self.sr = sr
         self.shape = shape
 
-    def coos(self) -> SparseCOOTensorReadIter:
```

Comment:
why did you delete `coos()`?  This should remain, as discussed with Seth.  It does not suffer from the same "scales with size of domain" issue that csr/csc have....

Reply:
@bkmartinjr it uses `.shape`:

```diff
 -        assert query.X("raw").coos().concat() == pa.SparseCOOTensor.from_numpy(
-            raw_X["soma_data"].to_numpy(),
-            np.array(
-                [
-                    raw_X["soma_dim_0"].to_numpy(),
-                    raw_X["soma_dim_1"].to_numpy(),
-                ]
-            ).T,
-            shape=soma_experiment.ms["RNA"].X["raw"].shape,
-        )
- 
-```

-- we can keep it but we'll need a bit of a refactor I think (or, store current not-necessarily-tight bounding box as array metadata ...)

Reply:
it is fine if it uses shape -- why is this a problem?

I suspect you are misunderstanding why use of `.shape` is problematic.  I'm happy to do a IRT discussion to detail it if useful.

Reply:
@bkmartinjr I did indeed misunderstood:

Before, my (clearly mistaken) belief from our conversation was "Uses `.shape`? Let it go."

Now, I know this proof of concept does work quite well:

```
#!/usr/bin/env python

import tiledbsoma as soma
import pyarrow as pa

exp = soma.Experiment.open("/Users/johnkerl/s/v/ileum2")

x = exp.ms["RNA"].X["data"]

arrow_table = x.read().tables().concat()

coo_data = arrow_table.column("soma_data").to_numpy()

coo_coords = np.array([
    arrow_table.column(f"soma_dim_{n}").to_numpy() for n in range(len(x.shape))
]).T

scoot = pa.SparseCOOTensor.from_numpy(coo_data, coo_coords, shape=x.shape)
```

even with

```
>>> x.shape
(9223372036854773760, 9223372036854773760)
>>> scoot.shape
(9223372036854773760, 9223372036854773760)
```

So I can confidently revert all the redlines of the COO bits on this PR.

Reply:
COO read-iterator restored on latest commit